### ISSUE TITLE
feature request: more flexible inject poison

### ISSUE THREAD

**Issue by schmkr** [h2non/toxy#26]
Right now, when using the inject poison, you need to inject all three `status`, `body` and `headers`. I'd love to be able to only inject one or two of them.

Furthermore, it would be very useful if the inject poison would merge injected headers with the original response headers. Right now, these headers are disregarded, resulting in e.g. missing CORS headers. Any change this could be added?

**Comment by schmkr** [comment]
Excuse me, I just noticed it is possible to omit the status code, resulting to a default `500`.

**Comment by h2non** [comment]
All the params `object` to the poison are actually optional. The main reason to encapsulate the arguments in a function of one arity is mainly to avoid passing multiple argument when they are optional.

Regarding to the headers feature you're proposing, I can easily merge already defined headers in the response, but not that if you're using this poison, the target server **never will be reached**, as this poison intercepts the incoming traffic in a blocking way. 
Probably I should add some flag in the docs to clarify if the poison is blocking or not.

Not that you can plug in your custom middleware to define CORS headers, for instance:
```js
var toxy = require('toxy')
var proxy = toxy()

proxy.use(function (req, res, next) {
  res.setHeader('Access-Control-Allow-Origin', '*')
  next()
})

proxy
  .all('/*')
  .poison(toxy.poisons.inject({ code: 503 })
  .withRule(toxy.rules.method('GET'))
```

**Comment by schmkr** [comment]
Thanks for the quick reply. That makes sense, that the target server is never reached. Never thought about that. Maybe there should be a new poison (e.g. `transform`) that does go through to the backend server, but after that, transforms the responses.?

I already worked around the issue of the cors headers just like you're describing. thanks.

**Comment by h2non** [comment]
I would consider adding a new poison to achieve that in a simple way, however it worth to say that you can actually achieve that using the [response middleware phase](https://github.com/h2non/rocky#types-of-middleware), which is provided natively by [`rocky`](https://github.com/h2non/rocky).  Also you might be interested in [`toxy.transformResponse()`](https://github.com/h2non/rocky#routetransformresponsemiddleware--filter-) built-in helper, which it's also an inherited feature from `rocky`.

Via the `response` middleware layer you can consistently intercept the response from your target server, and behave accordingly as you need. Here is pretty simple example replacing the headers and replying with a custom error, as you can do with the `inject` poison:
```js
var toxy = require('toxy')
var proxy = toxy()

// Intercept the response from the target server
proxy.useResponse(function (req, res, next) {
  if (res.getHeader('server') === 'nginx') {
    // Actually, this is your poison now :D
    res.setHeader('server', 'toxy')
    res.writeHead(500, res.headers)
    return res.end('Oh no!') // and we stop here!
  }
  next() // otherwise, continue with the next middleware
})

// Handle all the traffic
proxy
  .all('/*')
```

**Comment by h2non** [comment]
@schmkr FYI I've just released toxy `0.3.0`. A couple of notes below that you probably worth to know.

That release includes [two phases](https://github.com/h2non/toxy#poisoning-phases) for poisoning, one applied to the incoming traffic (like you can do in previous versions), and a new phase to poison the outgoing traffic (when request is received from the server in the proxy, and before it's sent to the client). 

Now you can plugin poisons in one or other stage according to your specific scenario. 
For instance, you can plug in `inject` as outgoing poison, enabling it based on the response status code, response headers or even response body.
Additionally, I've created new rules specifically for outgoing poisons. Take a look to the [API docs](https://github.com/h2non/toxy#response-headers).

**Comment by h2non** [comment]
Now headers [are merged](https://github.com/h2non/toxy/blob/master/lib/poisons/inject.js#L11) in the `inject` poison. You should just overwrite them.
